---
title: 'ABOUT'

category: General
layout: null
type: null
---
**Cellar** is an interactive tool for analyzing single-cell omics data. Cellar
is built in Python using the [Dash](https://plotly.com/dash/) framework
and relies on several open-source packages.

The app is developed and actively maintained by the
[Systems Biology Group](http://www.sb.cs.cmu.edu/) at
[Carnegie Mellon University](https://www.cmu.edu/). Our web-server
running Cellar can be accessed
[here](https://data.test.hubmapconsortium.org/app/cellar). See below
for a local installation. The source code is available on
[GitHub](https://github.com/ferrocactus/CellarV).

An accompanying preprint and supplementary files can be accessed through
[bioRxiv](https://www.biorxiv.org/content/10.1101/2021.03.19.436162v1?rss=1).

This [documentation](https://jingtaowang22.github.io/cellar_docs)
includes details on how to use Cellar and the data types
it supports. These include but are not limited to scRNA-seq, scATAC-seq,
CODEX, SNARE-seq, sciRNA-seq, Visium, CITE-seq. Cellar supports preprocessing,
dimensionality reduction, clustering, DE gene testing, enrichment analysis,
cluster and gene visualization modules, projection to spatial tiles,
label transfer, and semi-supervised clustering among other tools.
[Video tutorials](https://www.youtube.com/playlist?list=PL5sLSLkTYpWgfBQ0M8ObfBIqDMAzx0-D2)
are also available.

To install Cellar locally, pull the docker image with
```
docker pull euxhen/cellarv
```
and run it with
```
docker run --rm -p 8050:8050 euxhen/cellarv
```

After a short while, visit `localhost:8050` to view the app. For a list
of dependencies, please check the
[GitHub repository](https://github.com/ferrocactus/CellarV).