---
title: 'TUTORIAL: Integration of CITE-seq and CODEX Protein Data'

category: Tutorials
layout: null
type: tutorial
---
Cellar also supports CITE-seq data. [CITE-seq](https://www.nature.com/articles/nmeth.4380) measures the epitome and
transcriptome levels at the single cell level. To upload a CITE-seq dataset to Cellar, the data
must be in AnnData format, with `adata.X` corresponding to the mRNA expression matrix
and `adata.obsm['proteins']` corresponding to the protein expression matrix.
The names of the proteins must be stored under `adata.uns['proteins']`.

The expression levels of a protein can be visualized by selecting it from
the `Other Features` tab next to `Feature Visualization`.

<br>
<img src="images/citeseq-other.png" alt="drawing" class="w400"/>
<br>

Cellar also supports the integration of CITE-seq and CODEX protein data
using the [STvEA](https://www.science.org/doi/10.1126/sciadv.abc5464)
package which uses an anchor correction method to map features between the two.

To demonstrate this functionality, we have uploaded two test datasets to Cellar.
The CITE-seq dataset we use consists of [mRNA and protein expression
data from mice](https://www.biorxiv.org/content/10.1101/672501v1) and
the CODEX dataset profiles the [protein expression of tissue sections from
3 normal BALBc mice](https://www.cell.com/cell/fulltext/S0092-8674(18)30904-8).
To speed up calculations, we limit the analysis to only the first 3000 cells
of the CODEX dataset. The two datasets (`Govek_Troisi_CITE-seq` and `BALBc1_Goltsev_3000_CODEX`)
can be loaded from Cellar under the <span class="mbutton">DUAL MODE</span>.

<br>
<img src="images/cite-codex-main.png" alt="drawing" class="w600"/>
<br>

To perform the integration, first make sure the CODEX dataset is the active
one. Then under the `Tools` menu, select STvEA next to `Integrate` (image below). Several
settings related to data clean-up can be set along with the model used to
remove noise. We recommend cleaning both datasets since STvEA works best
if the expression values are scaled within the (0, 1) interval.

<br>
<img src="images/stvea.png" alt="drawing" class="w500"/>
<br>

Running STvEA might take a few minutes (fitting a negative binomial model
might take longer). Once the integration is complete, you can color the
CODEX data by the expression level of any gene that was mapped from the
CITE-seq expression levels under the `Other Features` tab.

<br>
<img src="images/codex-gene.png" alt="drawing" class="w400"/>
<br>
