---
title: 'LOAD DATA'

category: User Interface
layout: null
type: ui
---
Several datasets can be accessed from our website running Cellar at
[data.test.hubmapconsortium.org/app/cellar](https://data.test.hubmapconsortium.org/app/cellar).
Most of these are provided by [HuBMAP](https://hubmapconsortium.org/) and are
organized by data type, data center, and tissue. Data types include scRNA-seq,
CODEX, scATAC-seq, SNARE-seq, etc, while tissues include lymph, spleen, thymus,
kidney, heart, and more.

Datasets can be selected by first clicking **<span class='mbutton'>LOAD DATA</span>**
from the navigation bar which will expand the data panel. The select menu
lists all the uploaded datasets, followed by any Cellar datasets.

<br>
![CODEX Main Plot](images/codex-load-dataset.png)
<br>


* **<span class='mbutton'>UPLOAD</span>**
    * The default file format that Cellar uses is the
    [AnnData](https://anndata.readthedocs.io/en/latest/) or
    <span class='extension'>h5ad</span> file. We also accept
    <span class='extension'>csv</span> files or feature barcode sequences in
    [cellranger](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices) or
    [spaceranger](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/output/matrices)
    format as a <span class='extension'>gz</span> file.
    <br>
    <span class="warn">!!<span class="tooltip">Attention</span></span> In the case of
    <span class='extension'>h5ad</span> or <span class='extension'>csv</span> files,
    please make sure the data follows the cell $$\small\times $$ feature format,
    i.e., rows of the matrix correspond to cells and columns to features (genes, proteins).
    Additionaly, row and columns names must be provided with the file.
        * Example cell barcode: <span class='keyword'>TTACCATTCCAGCACG</span>. Example gene name (can be in ensembl format or HGNC symbol):
        <span class='keyword'>ENSG00000211699</span> or <span class='keyword'>TRGV3</span>.

    * We recommend uploading the data as an <span class='extension'>h5ad</span> file.
    An AnnData object stores expression matrices and annotations efficiently while also providing convenient methods for accessing
    and storing any additional unstructured objects. Furthermore, an AnnData object can be highly compressed, thus saving precious upload (download) time.
    Below we provide a Python snippet that can convert a <span class='extension'>csv</span> file into a maximally compressed AnnData object that
    can be later uploaded to Cellar. All you need is the `anndata` Python package that can be installed via `pip install anndata`. The following code
    reads a csv file `path/to/your/csv/file.csv` and writes it to another file `new/file/name.h5ad`.
    ```
    import anndata
    adata = anndata.read_csv('path/to/your/csv/file.csv')
    adata.write('new/file/name.h5ad', compression=9)
    ```
    The file can now be uploaded to Cellar. This procedure may reduce a dataset's size from several GBs to just a few MBs.

    * There is a limit of 3GB for any uploaded dataset. If you have a larger dataset,
    we recommend installing Cellar locally (follow instructions in the About tab or GitHub page),
    thus taking advantage of any computational resources you may have and avoiding network latency.

    * Uploaded datasets will appear first in the dropdown menu as `uploaded / datasetname`.

* **<span class='mbutton'>LOAD</span>**
    * This will load the selected dataset into memory which may
    take a few seconds depending on its size.
    * Once a dataset is loaded, the number of samples (cells) and the number of
    features will be shown on the top left corner of the main body.
    If the file contains 2D embeddings for the cells or any cluster
    assignments (all server datasets do), a scatter plot of the data
    will get shown automatically.
    Otherwise, running preprocessing (optional) and dimensionality reduction (see below)
    is typically the first step of the pipeline.
